src/gpodder/plugins/itunes.py

```diff
@@ -69,6 +71,30 @@ def __init__(self):
         self.priority = directory.Provider.PRIORITY_SECONDARY_SEARCH
 
     def on_search(self, query):
-        json_url = 'https://itunes.apple.com/search?media=podcast&term={}'.format(urllib.parse.quote(query))
-
-        return [directory.DirectoryEntry(entry['collectionName'], entry['feedUrl'], entry['artworkUrl100']) for entry in util.read_json(json_url)['results']]
+        offset = 0
+
+        while True:
+            json_url = f'https://itunes.apple.com/search?media=podcast&term={urllib.parse.quote(query)}&limit={PAGE_SIZE}&offset={offset}'
+            json_data = util.read_json(json_url)
+
+            if json_data['resultCount'] > 0:
+                for entry in json_data['results']:
+                    if entry.get('feedUrl') is None:
+                        continue
+
+                    title = entry['collectionName']
+                    url = entry['feedUrl']
+                    image = entry['artworkUrl100']
+
+                    yield(directory.DirectoryEntry(title, url, image))
+                    returned_res += 1
+
+                offset = offset + json_data['resultCount']
+            else:
+            '''
+            Unlike the podverse stop condition where we detect a resultCount smaller than the page size for apple we can only stop when 0 results
+            are returned because the API seems to consistently return more than the page size and does this in an inconsistent fasion, most often
+            returning 210 results but based on my observartion any number between page size and page size + 10 is possible.
+            With an API that does not obey its own rules the only valid stop condition is no results.
+            '''
```

Comment:
This should be a code comment, not a string. Also, the lines seem overly long, feel free to wrap it to 80 or 100 columns or so.

Reply:
I was under the impression that `'''` is the accepted style for multiline commenting/documentation functions/code these days (at least in some ansible stuff I contributed to).

As for wrapping it is roughly as long as some of the longer lines that contain URLs (139-144 vs 146), I kind of struggle with this, on my screen this fits with ease and could be double as long, I know there is discussion about this limit in more places and I really don't know where to take it.

Reply:
I guess PEP8 says to use `#` but then there is also the concept of docstrings and what Guido tweeted... :man_shrugging: 

It also says 79 char line length and spaces and not tabs (I really would like to switch to tabs due to the accessibility implication for visually impaired coders but am holding of because it is not the norm in python).

Reply:
Apparently line length also has an accessibility impact on the visually impaired so I pushed a PEP8 commit to this branch.

https://discuss.python.org/t/pep8-and-line-length/83129/41